vortex-datafusion/src/persistent/metrics.rs

```diff
@@ -0,0 +1,141 @@
+use std::sync::Arc;
+use std::time::Duration;
+
+use datafusion_physical_plan::metrics::{
+    Count, ExecutionPlanMetricsSet, Gauge, Label as DatafusionLabel,
+    MetricValue as DatafusionMetricValue, MetricsSet, Time,
+};
+use datafusion_physical_plan::Metric as DatafusionMetric;
+use vortex_metrics::{DefaultTags, Metric, MetricId, Tags, VortexMetrics};
+
+pub static PARTITION_LABEL: &str = "partition";
+
+#[derive(Clone, Debug, Default)]
+pub struct VortexExecMetrics {
+    pub vortex: Arc<VortexMetrics>,
```

Comment:
very nitty nit - seems like we're always using `VortexMetrics` in an `Arc`, we can push that `Arc` one layer down so there's slightly less visual clutter, e.g.
```rust
#[derive(Clone)]
pub struct VortexMetrics {
    inner: Arc<MetricsInner>
}

struct MetricsInner {
    registry: MetricRegistry,
    default_tags: DefaultTags,
    children: RwLock<Vec<Arc<VortexMetrics>>>,
}

fn things_that_takes_metrics(metrics: VortexMetrics) { todo!() }
```